core/app/finders/spree/taxons/find.rb

```diff
@@ -0,0 +1,61 @@
+module Spree
+  module Taxons
+    class Find
+      def initialize(taxons, params)
+        @taxons   = taxons
+
+        @roots    = params[:roots]
+        @ids      = params[:ids].split(',') if params[:ids].present?
+        @taxonomy = params[:taxonomy_id]
+        @name     = params[:name]
+      end
+
+      def call
+        taxons = @taxons
+
+        taxons = by_ids(taxons)      if ids?
+        taxons = by_taxonomy(taxons) if taxonomy?
+        taxons = by_roots(taxons)    if roots?
+        taxons = by_name(taxons)     if name?
+
+        taxons
+      end
+
+      private
+
+      attr_reader :ids, :taxonomy, :roots, :name
+
+      def ids?
+        ids.present?
+      end
+
+      def taxonomy?
+        taxonomy.present?
+      end
+
+      def roots?
+        roots.present?
```

Comment:
@bbonislawski What do you think about a separate class for this type of checks? You can find a pseudo-code in the previous comments